concourse-webhook-validator/src/concourse_webhook_validator/__init__.py

```diff
@@ -227,127 +202,165 @@ def display_results(webhooks_dict:dict) -> int:
-                        default='',
-                        help="The base directory in which the pipeline configurations are stored",
-                        nargs=1)
-    parser.add_argument('--deployment',
-                        default='',
-                        help="The name of the Concourse deployment",
-                        nargs=1)
-    parser.add_argument('--team',
-                        default='',
-                        help="The team name that the pipeline is configured for",
-                        nargs=1)
+    parser.add_argument(
+        "pipeline",
+        default=None,
+        help="The name of the pipeline configuration to load",
+        nargs=1,
+    )
+    parser.add_argument(
+        "--list",
+        action="store_true",
+        help="Specifies that the provided input is a file containing a list of pipeline configurations, one per line",
+    )
+    parser.add_argument(
+        "--base-dir",
+        default="",
+        help="The base directory in which the pipeline configurations are stored",
+        nargs=1,
+    )
+    parser.add_argument(
+        "--deployment", default="", help="The name of the Concourse deployment", nargs=1
+    )
+    parser.add_argument(
+        "--team",
+        default="",
+        help="The team name that the pipeline is configured for",
+        nargs=1,
+    )
     args = parser.parse_args()
-    
+
     if len(args.base_dir) == 0:
-        pipeline_base_dir = ''
+        pipeline_base_dir = ""
     else:
         pipeline_base_dir = args.base_dir[0]
-    
+
     if len(args.deployment) == 0:
-        pipeline_deployment = ''
+        pipeline_deployment = ""
     else:
         pipeline_deployment = args.deployment[0]
-    
+
     if len(args.team) == 0:
-        pipeline_team = ''
+        pipeline_team = ""
     else:
         pipeline_team = args.team[0]
-    
-    pipeline_file = args.pipeline[0]
-    pipeline_file_path = os.path.join(pipeline_base_dir, pipeline_deployment, pipeline_team, pipeline_file)
 
-    webhooks_dict = {}
-    format_output(
-        f"Checking webhooks configuration: {colours.bold}{pipeline_file}{colours.end}"
-    )
-    pipeline_config = load_pipeline_config(pipeline_file_path)
-    if validate_pipeline_config(pipeline_config) == 0:
-        parse_pipeline_resources(pipeline_config, webhooks_dict)
-        if len(webhooks_dict) > 0:
-            parse_pipeline_jobs(pipeline_config, webhooks_dict)
-            validate_webhooks(webhooks_dict)
-            if display_results(webhooks_dict) == 0:
-                print()
-                format_output(
-                    "Webhooks configuration check completed successfully"
-                )
-            else:
-                format_output(
-                    "Validation failures were encountered",
-                    "error"
-                )
-                sys.exit(1)
+    if args.list:
+        pipelines_list = read_pipelines_list_from_file(args.pipeline[0])
```

Comment:
Should args.list passed into the read function rather than args.pipeline?

Reply:
Ah no I see, we'll still pass in pipeline in every case and the list property is just a boolean?

Reply:
args.list here is just a boolean and defines whether the `--list` argument has been passed. When enabled the positional argument, args.pipeline, will be the file, rather than a specific pipeline configuration.

E.g.
```
concourse-webhook-validator my-pipeline
```
Will give: `my-pipeline` as `args.pipeline[0]` and `False` as `args.list`

```
concourse-webhook-validator a-list-of-files --list
```
Will give: `a-list-of-files` as `args.pipeline[0]` and `True` as `args.list`
```